internal/commands/helpers/helpers.go

```diff
@@ -82,18 +83,24 @@ func EvaluateStarlarkConfig(ctx context.Context, path string, env map[string]str
 }
 
 func ReadCombinedConfig(ctx context.Context, env map[string]string) (string, error) {
-	yamlConfig, err := ReadYAMLConfig(".cirrus.yml")
-	if err != nil {
-		return "", err
+	// Here we read the .cirrus.yaml first so that if the error would arise
+	// and will be inspected it would indicate the preferable extension
+	yamlConfig, yamlErr := ReadYAMLConfig(".cirrus.yaml")
+	if yamlErr != nil {
+		yamlConfig, yamlErr = ReadYAMLConfig(".cirrus.yml")
 	}
 
-	starlarkConfig, err := EvaluateStarlarkConfig(ctx, ".cirrus.star", env)
-	if err != nil {
-		if os.IsNotExist(err) {
-			return yamlConfig, nil
-		}
-		return "", err
+	starlarkConfig, starlarkErr := EvaluateStarlarkConfig(ctx, ".cirrus.star", env)
```

Comment:
Do you still need to check for `os.IsNotExist(err)`?

Reply:
Indeed! In the current state it's possible to miss a `cirrus.yml` that fails to read due to e.g. invalid permissions.

Reply:
Fixed in https://github.com/cirruslabs/cirrus-cli/pull/176/commits/5a59107bdb995f0cc6d28dba4253526771ca79ac.